perf_test/Benchmark_Context.hpp

```diff
@@ -117,26 +117,48 @@ inline void add_benchmark_context(bool verbose = false) {
 }
 
 template <class FuncType, class... ArgsToCallOp>
-inline void register_benchmark(const char* name, FuncType func,
+inline auto register_benchmark(const char* name, FuncType func,
```

Comment:
@lucbv , @cwpearson , is the intent for Kokkos Kernels to use google benchmark through this Benchmark_Context header? If so, I think it would be good for register_benchmark to return the benchmark object so the user can make additional config changes to it if they want. I was disappointed to see that google benchmark did not let me override UseManualTime with UseRealTime, so I had to make a second register_benchmark function. UseRealTime seems like an attractive setting since it automates more of the tedious time measuring stuff.

Reply:
These look like convenience functions wrapped around the standard Google Benchmark registration. I like these changes you're suggesting.